build.gradle

```diff
@@ -21,17 +34,86 @@ subprojects {
         testCompile([
                 "org.hamcrest:hamcrest-all:1.3",
                 "org.mockito:mockito-all:1.10.19",
-                "org.slf4j:slf4j-simple:1.7.+",
-                "org.awaitility:awaitility:2.0.+",
+                "org.slf4j:slf4j-simple:$slf4jVersion",
+                "org.awaitility:awaitility:2.0.0",
                 "com.google.guava:guava-testlib:$guavaVersion",
         ])
         testCompile("junit:junit:4.12") {
             exclude group: 'org.hamcrest', module: 'hamcrest-core'
         }
         compile([
-                "org.slf4j:slf4j-api:1.7.+",
+                "org.slf4j:slf4j-api:$slf4jVersion",
                 "org.apache.kafka:kafka-clients:$kafkaVersion",
                 "com.google.guava:guava:$guavaVersion"
         ])
     }
+
+    //Some OSGi information
+    jar {
+        manifest {
+            version = project.version
+            symbolicName = "$project.group.$baseName"
+        }
+    }
+
+    uploadArchives {
+        repositories {
+            signing {
+                required { shouldSign }
```

Comment:
Would it be prudent to add '&& gradle.taskGraph.hasTask("uploadArchives")' here.